server/actions_specs.go

```diff
@@ -198,6 +203,49 @@ func (s *RegistryServer) getApiSpecRevision(ctx context.Context, name names.Spec
 	return message, nil
 }
 
+// GUnzippedBytes uncompresses a slice of bytes.
+func GUnzippedBytes(input []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(input)
+	zr, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+	return ioutil.ReadAll(zr)
+}
+
+// GetApiSpecContents handles the corresponding API request.
+func (s *RegistryServer) GetApiSpecContents(ctx context.Context, req *rpc.GetApiSpecContentsRequest) (*httpbody.HttpBody, error) {
+	var specName = strings.TrimSuffix(req.GetName(), "/contents")
+	var spec *rpc.ApiSpec
+	var err error
+	if name, err := names.ParseSpec(specName); err == nil {
+		spec, err = s.getApiSpec(ctx, name, rpc.View_FULL)
+	} else if name, err := names.ParseSpecRevision(specName); err == nil {
+		spec, err = s.getApiSpecRevision(ctx, name, rpc.View_FULL)
+	}
+	if err != nil {
+		return nil, err
+	}
+	if spec == nil {
+		return nil, invalidArgumentError(fmt.Errorf("failed to locate %q", req.GetName()))
```

Comment:
If we failed to locate the spec, a "not found" error would have been returned at line 227. Receiving a `nil` spec with `nil` error isn't defined behavior and means we introduced a bug. Given we don't know what happened, I think "internal" or "unknown" error would make more sense.

Am I misunderstanding the purpose of this being an "invalid argument" error or the reason for its message?

Reply:
changed to internalError()